File: CS/App_Code/CommentsStorage.cs
using System;
using System.Web;
using System.Collections.Generic;

public static class CommentsStorage {
    private static Dictionary<Tuple<object, string>, string> Comments {
        get {
            if (HttpContext.Current.Session["CommentsStorage"] == null)
                HttpContext.Current.Session["CommentsStorage"] = new Dictionary<Tuple<object, string>, string>();
            return HttpContext.Current.Session["CommentsStorage"] as Dictionary<Tuple<object, string>, string>;
        }
    }

    private static bool CommentExists(object key, string columnName) {
        var tkey = new Tuple<object, string>(key, columnName);

        if (!Comments.ContainsKey(tkey))
            return false;

        return Comments[tkey] != null;
    }

    public static void Set(object key, string columnName, string comment) {
        Comments[new Tuple<object, string>(key, columnName)] = comment;
    }

    public static string Get(object key, string columnName) {
        if (CommentExists(key, columnName))
            return Comments[new Tuple<object, string>(key, columnName)];
        else
            return null;
    }

    public static void Delete(object key, string columnName) {
        Comments.Remove(new Tuple<object, string>(key, columnName));
    }
}

File: VB/App_Code/CommentsStorage.vb
Imports System
Imports System.Web
Imports System.Collections.Generic

Public NotInheritable Class CommentsStorage

	Private Sub New()
	End Sub

	Private Shared ReadOnly Property Comments() As Dictionary(Of Tuple(Of Object, String), String)
		Get
			If HttpContext.Current.Session("CommentsStorage") Is Nothing Then
				HttpContext.Current.Session("CommentsStorage") = New Dictionary(Of Tuple(Of Object, String), String)()
			End If
			Return TryCast(HttpContext.Current.Session("CommentsStorage"), Dictionary(Of Tuple(Of Object, String), String))
		End Get
	End Property

	Private Shared Function CommentExists(ByVal key As Object, ByVal columnName As String) As Boolean
		Dim tkey = New Tuple(Of Object, String)(key, columnName)

		If Not Comments.ContainsKey(tkey) Then
			Return False
		End If

		Return Comments(tkey) IsNot Nothing
	End Function

	Public Shared Sub [Set](ByVal key As Object, ByVal columnName As String, ByVal comment As String)
		Comments(New Tuple(Of Object, String)(key, columnName)) = comment
	End Sub

	Public Shared Function [Get](ByVal key As Object, ByVal columnName As String) As String
		If CommentExists(key, columnName) Then
			Return Comments(New Tuple(Of Object, String)(key, columnName))
		Else
			Return Nothing
		End If
	End Function

	Public Shared Sub Delete(ByVal key As Object, ByVal columnName As String)
		Comments.Remove(New Tuple(Of Object, String)(key, columnName))
	End Sub
End Class